core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -128,14 +103,16 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
     Topic.prototype.deleteSubtopic = function(subtopicId, isNewlyCreated) {
       var subtopicDeleted = false;
       for (var i = 0; i < this._subtopics.length; i++) {
-        if ((this._subtopics[i].getId() === subtopicId)) {
+        if (this._subtopics[i].getId() === subtopicId) {
           // When a subtopic is deleted, all the skills in it are moved to
           // uncategorized skill ids.
-          var skillIds = this._subtopics[i].getSkillIds();
-          for (var j = 0; j < skillIds.length; j++) {
-            var skillId = skillIds[j];
-            if (this._uncategorizedSkillIds.indexOf(skillId) === -1) {
-              this._uncategorizedSkillIds.push(skillId);
+          var skills = this._subtopics[i].getSkillSummaries();
```

Comment:
The variable name should be skillSummaries

Reply:
Done.